src/battle/battle_icon.c

```diff
@@ -0,0 +1,245 @@
+ */
+u32 BattleIcon_GetNARCIndex(void)
+{
+    return NARC_INDEX_BATTLE__GRAPHIC__PL_BATT_OBJ;
+}
+
+void BattleIcon_MakeTypeSpriteTiles(SpriteRenderer *param0, SpriteGfxHandler *param1, NNS_G2D_VRAM_TYPE param2, int typeIndex, u32 param4)
+{
+    sub_0200CBDC(param0, param1, BattleIcon_GetNARCIndex(), BattleIcon_GetMoveTypeTiles(typeIndex), 1, param2, param4);
+}
+
+void BattleIcon_MakeTypeSpritePalette(SpriteRenderer *param0, SpriteGfxHandler *param1, NNS_G2D_VRAM_TYPE param2, u32 param3)
+{
+    sub_0200CC9C(param0, param1, BattleIcon_GetNARCIndex(), BattleIcon_GetMoveTypePaletteFileIndex(), 0, 3, param2, param3);
+}
+
+void sub_0207C9B0(PaletteData *param0, int param1, SpriteRenderer *param2, SpriteGfxHandler *param3, NNS_G2D_VRAM_TYPE param4, u32 param5)
+{
+    sub_0200CD7C(param0, param1, param2, param3, BattleIcon_GetNARCIndex(), BattleIcon_GetMoveTypePaletteFileIndex(), 0, 3, param4, param5);
+}
+
+void sub_0207C9EC(SpriteRenderer *param0, SpriteGfxHandler *param1, u32 param2, u32 param3)
+{
+    sub_0200CE0C(param0, param1, BattleIcon_GetNARCIndex(), sub_0207C924(), 1, param2);
+    sub_0200CE3C(param0, param1, BattleIcon_GetNARCIndex(), sub_0207C928(), 1, param3);
+}
+
+void sub_0207CA34(SpriteGfxHandler *param0, u32 param1)
+{
+    SpriteGfxHandler_UnloadCharObjById(param0, param1);
+}
+
+void sub_0207CA3C(SpriteGfxHandler *param0, u32 param1)
+{
+    SpriteGfxHandler_UnloadPlttObjById(param0, param1);
+}
+
+void sub_0207CA44(SpriteGfxHandler *param0, u32 param1, u32 param2)
+{
+    SpriteGfxHandler_UnloadCellObjById(param0, param1);
+    SpriteGfxHandler_UnloadAnimObjById(param0, param2);
+}
+
+CellActorData *sub_0207CA58(SpriteRenderer *param0, SpriteGfxHandler *param1, int param2, const SpriteTemplate *param3)
+{
+    CellActorData *v0;
+    SpriteTemplate v1;
+
+    v1 = *param3;
+    v1.plttIdx = BattleIcon_GetMoveTypePaletteIndex(param2);
+    v0 = SpriteActor_LoadResources(param0, param1, &v1);
+
+    return v0;
+}
+
+void sub_0207CA88(CellActorData *param0)
+{
+    sub_0200D0F4(param0);
+}
+
+/*
+ * Given a move class index, return the corresponding NARC member containing the icon for that move class.
+ * See also BattleIcon_GetNARCIndex.
+ */
+u32 BattleIcon_GetMoveClassTiles(int moveClassIndex)
+{
+    GF_ASSERT(moveClassIndex < NELEMS(sMoveClassIconTiles));
+    return sMoveClassIconTiles[moveClassIndex];
+}
+
+/*
+ * Given a move class index, return the corresponding palette index to use for that move class' icon.
+ * See also BattleIcon_GetNARCIndex and BattleIcon_GetMoveTypePaletteFileIndex.
+ */
+u8 BattleIcon_GetMoveClassPaletteIndex(int moveClassIndex)
+{
+    GF_ASSERT(moveClassIndex < NELEMS(sMoveClassIconPaletteIndex));
+    return sMoveClassIconPaletteIndex[moveClassIndex];
+}
+
+/*
+ * Returns the index of the NARC for pl_batt_obj.
+ * (This is identical to BattleIcon_GetNARCIndex).
+ */
+u32 BattleIcon_GetMoveClassPalettteFileIndex(void)
+{
+    return NARC_INDEX_BATTLE__GRAPHIC__PL_BATT_OBJ;
```

Comment:
**polish**: Bumping that this should come from the NAIX.

Reply:
this one is actually a NARC index instead of a member index + comes from the NARC constants header, I will rename this function to make that explicit

Reply:
Oh! I totally missed that. Good catch on renaming the function.